appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/model/util/ModelUtils.java

```diff
@@ -395,36 +371,31 @@ public static boolean isAnnotationNull(Annotation annotation) {
         if (annotation == null) {
             return true;
         }
-        boolean allNull = true;
         for (Method m : annotation.annotationType().getDeclaredMethods()) {
             if (m.getParameterCount() == 0) {
                 try {
                     Object value = m.invoke(annotation);
-                    if (value != null) {
```

Comment:
FYI: Main point was to change the `.cast(...)` to `(Type)value` style of cast  or `instanceof` where appropriate. Also values from annotation can never be `null` or a `Collection` type so I removed these checks. To be sure this is correct I added tests for this util method.